cmd/healthcheck/main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+)
+
+var (
+	errInvalidPort = errors.New("bad config: invalid port")
+)
+
+// ValidatePort checks if the port configuration is valid
+func ValidatePort(port string) error {
+
+	if portInt, err := strconv.Atoi(port); err == nil {
+		if portInt < 1 || portInt > 65536 {
+			return errInvalidPort
+		}
+	} else {
+		return errInvalidPort
+	}
+	return nil
+}
+
+func main() {
+	host := "127.0.0.1"      // default host
+	usedPort := "13133"      // default port
+	path := "/health/status" //default path
+	generateCmd := flag.NewFlagSet("generate", flag.ExitOnError)
+	port := generateCmd.String("port", usedPort, "Specify collector health-check port")
+	validationErr := ValidatePort(*port)
```

Comment:
I think this needs to happen after the flags are parsed.

Reply:
Thanks. Updated.